### ISSUE TITLE
Permission denied when using -T flag (ssh-audit installed by snap)

### ISSUE THREAD

**Issue by MMquant** [jtesta/ssh-audit#124]
ssh-audit installed with snap
```
ubuntu@ip-xx-xx-30-85:~$ snap info ssh-audit
name:      ssh-audit
summary:   ssh-audit
publisher: Joe Testa (positronsecurity)
store-url: https://snapcraft.io/ssh-audit
contact:   jtesta@positronsecurity.com
license:   MIT
description: |
  SSH server and client security configuration auditor.  Official repository:
  <https://github.com/jtesta/ssh-audit>
commands:
  - ssh-audit
snap-id:      fuKZFTS5PBD7Lz2T761PRMibsSqbSP8l
tracking:     latest/stable
refresh-date: today at 09:38 UTC
channels:
  latest/stable:    2.5.0-1 2021-08-26 (4) 6MB -
  latest/candidate: ↑                          
  latest/beta:      2.2.0-1 2020-03-13 (1) 9MB -
  latest/edge:      ↑                          
installed:          2.5.0-1            (4) 6MB -
```

Input file with SSH targets
```
ubuntu@ip-xx-xx-30-85:~$ ls -la ssh-parsed.txt
-rw-rw-r-- 1 ubuntu ubuntu 15 Oct  8 09:49 ssh-parsed.txt
ubuntu@ip-xx-xx-30-85:~$ cat ssh-parsed.txt
xx.xx.30.85:22
```

Running ssh-audit with -T flag gives permission denied error
```
ubuntu@ip-xx-xx-30-85:~$ ssh-audit -T ssh-parsed.txt
Traceback (most recent call last):
  File "/snap/ssh-audit/4/bin/ssh-audit", line 8, in <module>
    sys.exit(main())
  File "/snap/ssh-audit/4/lib/python3.8/site-packages/ssh_audit/ssh_audit.py", line 1047, in main
    aconf = process_commandline(out, sys.argv[1:], usage)
  File "/snap/ssh-audit/4/lib/python3.8/site-packages/ssh_audit/ssh_audit.py", line 684, in process_commandline
    with open(aconf.target_file, 'r', encoding='utf-8') as f:
PermissionError: [Errno 13] Permission denied: 'ssh-parsed.txt'
```
Running as sudo gives the same error
```
ubuntu@ip-xx-xx-30-85:~$ sudo ssh-audit -T ssh-parsed.txt
Traceback (most recent call last):
  File "/snap/ssh-audit/4/bin/ssh-audit", line 8, in <module>
    sys.exit(main())
  File "/snap/ssh-audit/4/lib/python3.8/site-packages/ssh_audit/ssh_audit.py", line 1047, in main
    aconf = process_commandline(out, sys.argv[1:], usage)
  File "/snap/ssh-audit/4/lib/python3.8/site-packages/ssh_audit/ssh_audit.py", line 684, in process_commandline
    with open(aconf.target_file, 'r', encoding='utf-8') as f:
PermissionError: [Errno 13] Permission denied: 'ssh-parsed.txt'
```

However passing target directly as an ssh-audit argument works 
```
ubuntu@ip-xx-xx-30-85:~$ ssh-audit xx.xx.30.85:22
# general
(gen) banner: SSH-2.0-OpenSSH_8.2p1 Ubuntu-4ubuntu0.3
(gen) software: OpenSSH 8.2p1
(gen) compatibility: OpenSSH 7.4+, Dropbear SSH 2018.76+
(gen) compression: enabled (zlib@openssh.com)
...
```

Linux information
```
ubuntu@ip-xx-xx-30-85:~$ uname -a
Linux ip-xx-xx-30-85 5.11.0-1019-aws #20~20.04.1-Ubuntu SMP Tue Sep 21 10:40:39 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by MMquant** [comment]
Apparmor is intalled and configured as
```
ubuntu@ip-xx-xx-13-41:~$ sudo apparmor_status
apparmor module is loaded.
33 profiles are loaded.
31 profiles are in enforce mode.
   /snap/snapd/13170/usr/lib/snapd/snap-confine
   /snap/snapd/13170/usr/lib/snapd/snap-confine//mount-namespace-capture-helper
   /usr/bin/man
   /usr/lib/NetworkManager/nm-dhcp-client.action
   /usr/lib/NetworkManager/nm-dhcp-helper
   /usr/lib/connman/scripts/dhclient-script
   /usr/lib/snapd/snap-confine
   /usr/lib/snapd/snap-confine//mount-namespace-capture-helper
   /usr/sbin/tcpdump
   /{,usr/}sbin/dhclient
   lsb_release
   man_filter
   man_groff
   nvidia_modprobe
   nvidia_modprobe//kmod
   snap-update-ns.amazon-ssm-agent
   snap-update-ns.lxd
   snap-update-ns.ssh-audit
   snap.lxd.activate
   snap.lxd.benchmark
   snap.lxd.buginfo
   snap.lxd.check-kernel
   snap.lxd.daemon
   snap.lxd.hook.configure
   snap.lxd.hook.install
   snap.lxd.hook.remove
   snap.lxd.lxc
   snap.lxd.lxc-to-lxd
   snap.lxd.lxd
   snap.lxd.migrate
   snap.ssh-audit.ssh-audit
2 profiles are in complain mode.
   snap.amazon-ssm-agent.amazon-ssm-agent
   snap.amazon-ssm-agent.ssm-cli
2 processes have profiles defined.
0 processes are in enforce mode.
2 processes are in complain mode.
   /snap/amazon-ssm-agent/4046/amazon-ssm-agent (1126) snap.amazon-ssm-agent.amazon-ssm-agent
   /snap/amazon-ssm-agent/4046/ssm-agent-worker (1155) snap.amazon-ssm-agent.amazon-ssm-agent
0 processes are unconfined but have a profile defined.
```
Disabling apparmor entirely (`sudo systemctl stop apparmor`) didn't work

Btw this is the Ubuntu machine used https://aws.amazon.com/marketplace/pp/prodview-zvdilnwnuopoo

Regarding your cat commands

```
ubuntu@ip-xx-xx-13-41:~$ cat ssh-parsed 
127.0.0.1:22
ubuntu@ip-xx-xx-13-41:~$ sudo cat ssh-parsed 
127.0.0.1:22
ubuntu@ip-xx-xx-13-41:~$ ssh-audit -T ssh-parsed
Traceback (most recent call last):
  File "/snap/ssh-audit/4/bin/ssh-audit", line 8, in <module>
    sys.exit(main())
  File "/snap/ssh-audit/4/lib/python3.8/site-packages/ssh_audit/ssh_audit.py", line 1047, in main
    aconf = process_commandline(out, sys.argv[1:], usage)
  File "/snap/ssh-audit/4/lib/python3.8/site-packages/ssh_audit/ssh_audit.py", line 684, in process_commandline
    with open(aconf.target_file, 'r', encoding='utf-8') as f:
PermissionError: [Errno 13] Permission denied: 'ssh-parsed'
ubuntu@ip-xx-xx-13-41:~$ sudo ssh-audit -T ssh-parsed
Traceback (most recent call last):
  File "/snap/ssh-audit/4/bin/ssh-audit", line 8, in <module>
    sys.exit(main())
  File "/snap/ssh-audit/4/lib/python3.8/site-packages/ssh_audit/ssh_audit.py", line 1047, in main
    aconf = process_commandline(out, sys.argv[1:], usage)
  File "/snap/ssh-audit/4/lib/python3.8/site-packages/ssh_audit/ssh_audit.py", line 684, in process_commandline
    with open(aconf.target_file, 'r', encoding='utf-8') as f:
PermissionError: [Errno 13] Permission denied: 'ssh-parsed'
```

**Comment by MMquant** [comment]
I simply tried to clone the github repo and -T flag works. There is a problem with the snap installation/snap permissions.

**Comment by jtesta** [comment]
@MMquant I published an updated Snap package (2.5.0-2).  I think updates are applied automatically every 6(?) hours, so you'll get it soon.  Or you can manually run `sudo snap refresh` to force an update.

Once you have 2.5.0-2 (you can check with `snap info ssh-audit`), you can optionally give it read & write permissions to your home directory with the following command: `sudo snap connect ssh-audit:home :home` (note that hidden files and directories would still be disallowed).  And either way, `$HOME/snap/ssh-audit/common/` will always be readable & writable.

Thanks for reporting this issue!  If you run into any more problems, please let me know!
